docs/reference/api-conventions.asciidoc

```diff
@@ -19,15 +19,59 @@ specified using the `source_content_type` query string parameter.
 {es} only supports UTF-8-encoded JSON. {es} ignores any other encoding headings
 sent with a request. Responses are also UTF-8 encoded.
 
+[discrete]
+=== X-Opaque-id requirements
+`X-opaque-id` - a header used by elasticsearch to allow identify and correlate
+the origin of the request with the {es} logs. It is emitted in deprecation
+logs as `elasticsearch.x-opaque-id` (both JSON log file and indexed logs)
+and is used together with `key` of the deprecation to deduplicate deprecation logs.
+
+See link:./server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java[RateLimitingFilter]
+which is used for both indexing deprecation logs and writing to a file.
+
+The use of `x-opaque-id` in deduplication can be turned off for indexed
+deprecation logs with the use of `cluster.deprecation_indexing.x_opaque_id_used.enabled`.
+This is intended to be used for small clusters which are limited in resources.
+
+This setting will not affect writing deprecation logs to file.
+
+It is emitted also in search slow logs as an `id` field and in audit logs as `opaque_id`.
+The value is also present in <<_identifying_running_tasks, `task api`>>.
+
+This header can contain any value, but should be from a finite set.
+If there is a lot of different `x-opaque-id` values, it will prevent deduplication from working
```

Comment:
This section currently duplicates content on other pages. I'd rather link to those pages so users can get more context.

```suggestion
[discrete]
[[x-opaque-id]]
=== `X-Opaque-Id` HTTP header

You can pass an `X-Opaque-Id` HTTP header to track the origin of a request in
{es} logs and tasks. If provided, {es} surfaces the `X-Opaque-Id` value in the:

* Response of any request that includes the header
* <<_identifying_running_tasks,Task management API>> response
* <<_identifying_search_slow_log_origin,Slow logs>>
* <<deprecation-logging,Deprecation logs>>

For the deprecation logs, {es} also uses the `X-Opaque-Id` value to throttle
and deduplicate deprecation warnings. See <<_deprecation_logs_throttling>>.

The `X-Opaque-Id` header accepts any arbitrary value. However, we recommend you
limit these values to a finite set. Don't generate a unique `X-Opaque-Id` header
for every request. Too many unique `X-Opaque-Id` values can prevent {es} from
deduplicating warnings in the deprecation logs.
```